okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpEngine.java

```diff
@@ -674,6 +674,7 @@ public void readResponse() throws IOException {
       return; // No network response to read.
     }
 
+    Request sentRequest = userRequest;
```

Comment:
Curious : do tests fail if you drop this field, and just use the initial `networkRequest` ?